dotnet-desktop-guide/net/winforms/migration/index.md

```diff
@@ -0,0 +1,231 @@
+
+      <PropertyGroup>
+        <OutputType>WinExe</OutputType>
+        <TargetFramework>net5.0-windows</TargetFramework>
+        <UseWindowsForms>true</UseWindowsForms>
+        <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+      </PropertyGroup>
+
+    </Project>
+    ```
+
+    > [!IMPORTANT]
+    > Libraries don't need to define an `<OutputType>` setting. Remove that entry if you're upgrading a library project.
+
+This XML gives you the basic structure of the project. However, it doesn't contain any of the settings from the old project file. Using the old project information you previously copied to a text editor, copy the following settings.
+
+01. With the `<PropertyGroup>` element in the new project, copy the following from the old project:
+
+    - `<RootNamespace>`
+    - `<AssemblyName>`
+
+    Your project file should look similar to the following:
+
+    ```xml
+    <Project Sdk="Microsoft.NET.Sdk">
+
+      <PropertyGroup>
+        <OutputType>WinExe</OutputType>
+        <TargetFramework>net5.0-windows</TargetFramework>
+        <UseWindowsForms>true</UseWindowsForms>
+        <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+
+        <RootNamespace>MatchingGame</RootNamespace>
+        <AssemblyName>MatchingGame</AssemblyName>
+      </PropertyGroup>
+
+    </Project>
+    ```
+
+01. After the `</PropertyGroup>` element closure, copy the `<ItemGroup>` entries from the old project that contain `<ProjectReference>` or `<PackageReference>`.
+
+    Your project file should look similar to the following:
+
+    ```xml
+    <Project Sdk="Microsoft.NET.Sdk">
+
+      <PropertyGroup>
+        (contains settings previously described)
+      </PropertyGroup>
+
+      <ItemGroup>
+        <ProjectReference Include="..\MatchingGame.Logic\MatchingGame.Logic.csproj">
+          <Project>{36b3e6e2-a9ae-4924-89ae-7f0120ce08bd}</Project>
+          <Name>MatchingGame.Logic</Name>
+        </ProjectReference>
+      </ItemGroup>
+      <ItemGroup>
+        <PackageReference Include="MetroFramework">
+          <Version>1.2.0.3</Version>
+        </PackageReference>
+      </ItemGroup>
+
+    </Project>
+    ```
+
+    The `<ProjectReference>` elements don't need the `<Project>` and `<Name>` children, you can remove those:
+
+    ```xml
+    <ItemGroup>
+      <ProjectReference Include="..\MatchingGame.Logic\MatchingGame.Logic.csproj" />
+    </ItemGroup>
+    ```
+
+### Resources and settings
+
+Windows Forms projects for .NET Framework typically include other files such as *Properties/Settings.settings* and *Properties/Resources.resx*. These files, and any *resx* file created for your app besides form *resx* files, would need to be migrated.
+
+Copy those entries from the old project file into an `<ItemGroup>` entry in the new project. After you copy the entries, change any `<Compile Include="value">` or `<EmbeddedResource Include="value">` to instead use `Update` instead of `Include`, otherwise you'll get a compile error.
```

Comment:
```suggestion
Copy those entries from the old project file into an `<ItemGroup>` element in the new project. After you copy the entries, change any `<Compile Include="value">` or `<EmbeddedResource Include="value">` elements to instead use `Update` instead of `Include`.
```